walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -11,13 +11,130 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
-
+ 
 package com.google.sps;
-
+ 
+import java.util.Arrays;
+import java.util.ArrayList;
 import java.util.Collection;
-
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+ 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    // create a full day of free timeslots
+    List<TimeRange> fullDay = new ArrayList<TimeRange>();
+    fullDay.add(TimeRange.WHOLE_DAY);
+    
+    // cannot modify arraylist in an enhanced for loop, must create separate arraylists
+    List<TimeRange> freeTimes = new ArrayList<TimeRange>();
+    List<TimeRange> blockedTimes = new ArrayList<TimeRange>();
+
+
+    Collection<String> attendees = request.getAttendees();
+    Set<String> eventAttendees;
+    
+  for(Event event: events){
+    eventAttendees = event.getAttendees();
+    boolean isOverlappedAttendee = false;
+    for(String attendee:eventAttendees){
+      if(attendees.contains(attendee)){
+        isOverlappedAttendee = true;
+        // add a break to terminate loop early if at least one attendee is present at event
+        break; 
+        }
+      }
+      if(isOverlappedAttendee){
```

Comment:
Your indentation is off for this section. Should be two spaces back

Reply:
ah sorry, I still have not setup a proper IDE that would tell me these things.